catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/security/UserApplication.java

```diff
@@ -55,6 +55,36 @@ public UserApplication(EndpointUtil util, PersistentStore persistentStore) {
         this.persistentStore = persistentStore;
     }
 
+    @Override
+    public void init() {
+        get("/user", (req, res) -> {
+            Subject subject = (Subject) SecurityUtils.getSubject();
+            res.type(APPLICATION_JSON);
+            return getSubjectAttributes(subject);
+        }, util::getJson);
+
+        put("/user/preferences", APPLICATION_JSON, (req, res) -> {
+            Subject subject = (Subject) SecurityUtils.getSubject();
+
+            if (subject.isGuest()) {
+                res.status(401);
+                return ImmutableMap.of("message", "Guest cannot save preferences.");
+            }
+
+            Map<String, Object> preferences = JsonFactory.create()
+                    .parser()
+                    .parseMap(util.safeGetBody(req));
+
+            if (preferences == null) {
+                preferences = new HashMap<>();
+            }
+
+            setUserPreferences(subject, preferences);
+
+            return preferences;
+        }, util::getJson);
```

Comment:
When the preference call fails with a 413, the message coming back is an `Issue Authorizing Request` and is misleading.  This was observed when the upload size was restricted to 10 bytes (which may not happen in reality, but still).

Reply:
Ahhhh you are correct. I forgot to add the error handling in this application. Good catch 👍

Reply:
Fixed